### PR TITLE
Reindex Basic implementation - Structure Code

### PR DESCRIPTION
**Pull request by pengchen0692**
## Description
Implement Reindex Operation. 
As a basic implemention, it allows reindex operation be triggered and execute until completion. 

## Related issues
Addresses [AB#83018](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/83018)

## Testing
All tests pass

### CODE REVIEW

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+/*************************************************************
+    Reindex State Table. 
+    Reindex state on each extended query tag.
+**************************************************************/
+IF NOT EXISTS (
+    SELECT * 
+    FROM sys.tables
+    WHERE name = 'ReindexState')
+BEGIN
+    CREATE TABLE dbo.ReindexState (
```
Comment (smithago) [inline src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql:18]: While discussing with Will, we decided we do not need this. All we need is Tag table index status.

Comment (smithago) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql:18]: Think like a framework who's concern is to handle the indexing internal state. Externaly all you care is passed or failed. To use it with QIDO.

Job framework also provides job status and error messages if failed

Comment (smithago) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql:18]: We should just add Operation ID To existing tag table

**Review by smithago — CHANGES_REQUESTED** [review]
why do we need the new table? Can you follow up with Will

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql (hunk 1 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql:18]: Thanks Smitha!
Discussed with Will. 
* How to store internal state is still in discussion.
* This PR would start with a simple solution , which will be incrementally improved later

**Comment by pengchen0692** [comment]
Due to https://microsofthealth.visualstudio.com/Health/_workitems/edit/81200, created new PR https://github.com/microsoft/dicom-server/pull/864

**Comment by pengchen0692** [comment]
Due to https://microsofthealth.visualstudio.com/Health/_workitems/edit/81200, created new PR #864

**Pull request closed without merging**